core/invoker/src/main/scala/whisk/core/containerpool/docker/DockerContainer.scala

```diff
@@ -73,6 +75,8 @@ object DockerContainer {
             case (key, value) => Seq("-e", s"$key=$value")
         }.flatten
 
+        val dnsArgs = dnsServer.foldLeft(Array.empty[String]) { (a,d) => a ++ Array("--dns", d) }
```

Comment:
Should be equivalent to:

```scala
dnsServers.map(Seq("--dns", _)).flatten
```

(I'd always prefer the more abstract `map` over a relatively lowlevel `foldLeft`).

Also note that `Array` is a mutable/indexed datastructure, thus it's usage should be chosen carefully.